lib/resources/user-preferences.js

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const { always } = require('ramda');
+const Problem = require('../util/problem');
+const { getOrNotFound } = require('../util/promise');
+
+module.exports = (service, endpoint) => {
+
+  //////////////////////////////////////////////////////////////////////////////
+  // User preferences (UI settings)
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Endpoint to get all of a user's preferences.
+  // For completeness and ease of debugging; as these preferences are normally
+  // pulled in by the frontend through the extended version of /users/current.
+  service.get('/user-preferences', endpoint.simple(({ UserPreferences }, { auth }) => {
+    if (auth.actor.value === undefined) return Problem.user.insufficientRights();
+    return UserPreferences.getForUser(auth.actor.value.id)
+      .then(res => ({ headers: { 'Content-Type': 'application/json; charset=utf-8' }, body: res.preferences }));
+  }));
```

Comment:
Can we not add this and just use the `/v1/users/current` endpoint to test? If we dont add it, we don't have to document it 😅